src/borg/locking.py

```diff
@@ -158,6 +161,41 @@ def is_locked(self):
     def by_me(self):
         return os.path.exists(self.unique_name)
 
+    def kill_dead_lock(self):
+        for name in os.listdir(self.path):
+
+            try:
+                host_pid, thread = name.rsplit('-', 1)
+                host, pid = host_pid.rsplit('.', 1)
+            except ValueError:
+                # Malformed lock name? Or just soem new format we don't understand?
+                # It's safer to just exit
+                return False
+
+            if host != _hostname:
+                return False
+
+            try:
+                # This may not work in Windows.
+                os.kill(int(pid), 0)
+                return False
+            except OSError:
+                pass
+
+            # We should be ok not checking thread no matter what it is, right?
+            # Possible implications on Windows?
+
+            if not self.ok_to_kill_zombie_locks:
+                print(("Found zombie lock %s, but not deleting because repo option uniq_hostname is not set" % name))
+                return False
+
+            os.unlink(os.path.join(self.path, name))
+
+        # If there's a race and some other locker appeared, we'd get an exception
+        # Do we need to handle it, and if so - how?
+        os.rmdir(self.path)
```

Comment:
iirc, the lock is the DIRECTORY. The file inside is just informative.
So, a successful unlink is releasing the lock.